docker-app/qfieldcloud/core/tests/test_secret.py

```diff
@@ -0,0 +1,179 @@
+        s1_assigned = self._create_secret(
+            name="s1", project=self.p1, assigned_to=self.u1
+        )
+        s2_assigned = self._create_secret(
+            name="s2", project=self.p1, assigned_to=self.u1
+        )
+        s3_project = self._create_secret(name="s3", project=self.p1)
+        _s1_another_project = self._create_secret(
+            name="s1", project=self.p2, assigned_to=self.u1
+        )
+        _s1_another_user = self._create_secret(
+            name="s1", project=self.p1, assigned_to=self.u2
+        )
+
+        secrets = Secret.objects.for_user_and_project(self.u1, self.p1)
+
+        self.assertEqual(secrets.count(), 3)
+        self.assertEqual(secrets[0], s1_assigned)
+        self.assertEqual(secrets[1], s2_assigned)
+        self.assertEqual(secrets[2], s3_project)
+
+    def test_add_project_secret_available_for_owner(self):
+        s1 = self._create_secret(name="s1", project=self.p3)
+
+        secrets = Secret.objects.for_user_and_project(self.u1, self.p3)
+        self.assertEqual(secrets.count(), 1)
+        self.assertEqual(secrets[0], s1)
+
+    def test_assigned_secret_overrides_ind_project_secret(self):
+        _s1_project = self._create_secret(name="s1", project=self.p3)
+        s1_assigned = self._create_secret(
+            name="s1", project=self.p3, assigned_to=self.u1
+        )
+
+        secrets = Secret.objects.for_user_and_project(self.u1, self.p3)
+
+        self.assertEqual(secrets.count(), 1)
+        self.assertEqual(secrets[0], s1_assigned)
+
+    def test_add_secrets_to_project_and_user_overrides(self):
+        p = Project.objects.create(name="project", owner=self.u1)
+        self._create_secret(name="s1", project=p)
+        s1_assigned = self._create_secret(name="s1", project=p, assigned_to=self.u1)
+
+        secrets = Secret.objects.for_user_and_project(self.u1, p)
+
+        self.assertEqual(secrets.count(), 1)
+        self.assertEqual(secrets[0], s1_assigned)
+
+    def test_add_secrets_to_organization_and_project_overrides(self):
+        self._create_secret(name="s1", organization=self.o1)
+
+        p = Project.objects.create(name="project", owner=self.u1)
+        s1_project = self._create_secret(name="s1", project=p)
+
+        secrets = Secret.objects.for_user_and_project(self.u1, p)
+
+        self.assertEqual(secrets.count(), 1)
+        self.assertEqual(secrets[0], s1_project)
+
+    def test_check_secrets_type_priority(self):
+        p = Project.objects.create(name="project", owner=self.u1)
+
+        self._create_secret(name="s1", organization=self.o1)
+        self._create_secret(name="s1", organization=self.o1, assigned_to=self.u1)
+        self._create_secret(name="s1", project=p)
+        s1_assigned = self._create_secret(name="s1", project=p, assigned_to=self.u1)
+
+        secrets = Secret.objects.for_user_and_project(self.u1, p)
+
+        self.assertEqual(secrets.count(), 1)
+        self.assertEqual(secrets[0], s1_assigned)
```

Comment:
```suggestion
        secrets = Secret.objects.for_user_and_project(self.u1, p)
        
        s1_org = self._create_secret(name="s1", organization=self.o1)

        secrets.refresh_from_db()
        self.assertEqual(secrets.count(), 1)
        self.assertEqual(secrets[0], s1_org)

        s1_org_assigned = self._create_secret(name="s1", organization=self.o1, assigned_to=self.u1)

        secrets.refresh_from_db()
        self.assertEqual(secrets.count(), 1)
        self.assertEqual(secrets[0], s1_org_assigned)

        s1_project = self._create_secret(name="s1", project=p)

        secrets.refresh_from_db()
        self.assertEqual(secrets.count(), 1)
        self.assertEqual(secrets[0], s1_project)

        s1_project_assigned = self._create_secret(name="s1", project=p, assigned_to=self.u1)

        secrets.refresh_from_db()
        self.assertEqual(secrets.count(), 1)
        self.assertEqual(secrets[0], s1_project_assigned)
```

Since we have the test, I think we can check the priority for each step.